src/models/liturgical-day/liturgical-day.types.ts

```diff
@@ -0,0 +1,208 @@
+     */
+    liturgicalColors?: RomcalLiturgicalColor | RomcalLiturgicalColor[];
+    /**
+     * Cycle metadata of a liturgical day.
+     */
+    cycles?: Partial<RomcalCyclesMetadata>;
+    /**
+     * The source of the liturgical day.
+     *
+     * This value is used when localizing dates so that the [[Locales.localizeDates]] function knows
+     * which subtree in the locale file to look for.
+     *
+     * A special key, "temporal", may be used when you do not wish to specify any subtree but rather
+     * provide the entire path (as a period delimited string) to the [[Locales.localizeDates]] function
+     * to lookup.
+     */
+    source?: RomcalLiturgicalDaySources;
+    /**
+     * If this liturgical day must be removed from this calendar and from all those it inherits,
+     * on the final calendar generated by romcal.
+     */
+    drop?: boolean;
+    /**
+     * Allow to extend or replace properties of an already defined liturgical day (that have the same key).
+     */
+    extend?: boolean;
+  };
+
+export type LiturgicalDayExtendedInput = Omit<BaseLiturgicalDayInput, 'date' | 'extend'> &
+  Partial<Pick<BaseLiturgicalDayInput, 'date'>> &
+  Required<Pick<BaseLiturgicalDayInput, 'extend'>>;
+
+/**
+ * This type is used internally by romcal
+ * during the construction of the liturgical calendar.
+ * All properties except `key` and `date` are marked optional as
+ * the the object is constructed in stages. This interface
+ * should not be used in consumer applications.
+ */
+export type LiturgicalDayInput = BaseLiturgicalDayInput | LiturgicalDayExtendedInput;
+
+export type LiturgyDayDiff = Partial<Pick<LiturgicalDayInput, 'name' | 'rank' | 'prioritized' | 'liturgicalColors'>> & {
+  metadata?: Partial<LiturgicalDayMetadata>;
+};
+
+export type LiturgyDayExtendedMetadata = {
+  fromCalendar: RomcalCalendarName;
+  diff: LiturgyDayDiff;
+};
+
+export type RomcalCalendarMetadata = {
+  totalWeeksInGregorianYear: number;
+  totalWeeksInLiturgicalYear: number;
+  weekOfGregorianYear: number;
+  weekOfLiturgicalYear: number;
+  weekOfSeason: number;
+  dayOfGregorianYear: number;
+  dayOfLiturgicalYear: number;
+  dayOfSeason: number;
+  dayOfWeek: number; // 0 | 1 | 2 | 3 | 4 | 5 | 6; // 0 is Sunday, 6 is Saturday
+  dayOfWeekCountInMonth: number;
+  startOfLiturgicalYear: Date | string;
+  endOfLiturgicalYear: Date | string;
+  easter: Date | string;
+};
+
+export type LiturgicalDayMetadata = {
+  titles: RomcalTitle[];
+};
+
+/**
+ * The source of the liturgical day.
+ *
+ * This value is used when localizing dates so that the [[Locales.localizeDates]] function knows
+ * which subtree in the locale file to look for.
+ *
+ * A special key, "temporal", may be used when you do not wish to specify any subtree but rather
+ * provide the entire path (as a period delimited string) to the [[Locales.localizeDates]] function
+ * to lookup.
+ */
+export type RomcalLiturgicalDaySources =
+  | 'temporal'
```

Comment:
This is part of a probably deprecated code, and must be removed as soon as I have the time to double check.

Reply:
If so, I think it should be done before v2 release, or even better: in this very PR.

Reply:
Before v2 release I agree. In this PR I'm not sure. I hope to have a beta release asap, but if we always add new things to do, it might never happen 😄

Reply:
Okay. Will you keep this in your personall TODO list or should I open an issue for all these potentially deprecated code portions?